sydent/http/servlets/emailservlet.py

```diff
@@ -125,18 +144,19 @@ def do_validate_request(self, request):
             }
 
         try:
-            resp = self.sydent.validators.email.validateSessionWithToken(sid, clientSecret, tokenString)
+            return self.sydent.validators.email.validateSessionWithToken(sid, clientSecret, tokenString)
         except IncorrectClientSecretException:
-            return {'success': False, 'errcode': 'M_INCORRECT_CLIENT_SECRET',
+            return {'success': False, 'errcode': 'M_INVALID_PARAM',
```

Comment:
This doesn't feel right?

Reply:
`M_INCORRECT_CLIENT_SECRET` doesn't exist in the spec.

Reply:
Ugh, do you know if clients use it at all?

Reply:
Neither Riot web, ios or android uses it.

Reply:
Ok, cool. Might be worth mentioning it in #synapse-dev:matrix.org, but should be fine

Reply:
Done.